src/v2/Components/UserSettings/TwoFactorAuthentication/Components/BackupSecondFactor/__tests__/BackupSecondFactorActions.jest.tsx

```diff
@@ -0,0 +1,73 @@
+import { mount } from "enzyme"
+import { BackupSecondFactorActions } from "../BackupSecondFactorActions"
+
+describe("Two factor authentication enrollment", () => {
+  const props = {
+    backupSecondFactors: ["d3bd78d468", "7aa4c5922c"],
+  }
+  const getWrapper = (passedProps = props) => {
+    return mount(<BackupSecondFactorActions {...passedProps} />)
+  }
+
+  const getCopyButton = () => {
+    return getWrapper().find('[data-test="copyButton"]').first()
+  }
+
+  describe("when the browser does not support clipboard", () => {
+    it("displays a copy button based on browser support", () => {
+      expect(getCopyButton()).toHaveLength(0)
+    })
+  })
+
+  describe("when the browser supports clipboard", () => {
+    const mockClipboard = { writeText: jest.fn() }
+    beforeEach(() => {
+      Object.assign(navigator, {
+        clipboard: mockClipboard,
+      })
+    })
+
+    it("displays a copy button", () => {
+      const copyButton = getCopyButton()
+
+      expect(copyButton).toHaveLength(1)
+      expect(copyButton.text()).toBe("Copy")
+    })
+
+    it("enables user to copy the recovery codes", () => {
+      const copyButtonProps = getCopyButton().props()
+
+      if (copyButtonProps.onClick) copyButtonProps.onClick({} as any)
+
+      expect(navigator.clipboard.writeText).toBeCalledTimes(1)
+    })
+  })
+
+  describe("txt file download", () => {
+    const downloadButton = getWrapper()
+      .find('[data-test="downloadButton"]')
+      .first()
+
+    it("displays a download button", () => {
+      expect(downloadButton).toHaveLength(1)
+      expect(downloadButton.text()).toBe("Download")
+    })
+
+    it("enables user to download the recovery codes", () => {
+      global.URL.createObjectURL = jest.fn()
+
+      // @ts-ignore: Type
+      global.Blob = function (content, options) {
```

Comment:
note its pretty rare to use `function` keyword like this with arrow functions now being the standard: 

```tsx
global.Blob = (content, options) => { ... } 
```